objects/src/transaction/prepared_tx.rs

```diff
@@ -54,19 +53,24 @@ impl PreparedTransaction {
 
     /// Return a reference the transaction script.
     pub fn tx_script(&self) -> Option<&TransactionScript> {
-        self.tx_script.as_ref()
+        self.tx_args.tx_script()
     }
 
     /// Returns a reference to the inputs for this transaction.
     pub fn tx_inputs(&self) -> &TransactionInputs {
         &self.tx_inputs
     }
 
+    /// Return a reference the transaction script.
+    pub fn note_args(&self) -> Option<&BTreeMap<NoteId, Word>> {
+        self.tx_args.note_args()
+    }
```

Comment:
Similar comment as above: I wonder if we should combine `tx_script()` and `note_args()` accessors into a single `tx_args()` accessor.